neqo-transport/src/connection.rs

```diff
@@ -1561,26 +1676,106 @@ impl Connection {
                 }
             }
         } else {
-            debug_assert_eq!(self.role, Role::Server);
             Ok(())
         }
     }
 
-    /// Process the final set of transport parameters.
-    fn process_tps(&mut self) -> Res<()> {
-        self.validate_odcid()?;
-        if let Some(token) = self
-            .tps
-            .borrow()
-            .remote
-            .as_ref()
-            .unwrap()
-            .get_bytes(tparams::STATELESS_RESET_TOKEN)
-        {
-            let reset_token = <[u8; 16]>::try_from(token).unwrap().to_owned();
-            self.path.as_mut().unwrap().set_reset_token(reset_token);
+    fn validate_cids_draft_28_plus(&mut self) -> Res<()> {
+        let tph = self.tps.borrow();
+
+        let tp = tph
+            .remote()
+            .get_bytes(tparams::INITIAL_SOURCE_CONNECTION_ID);
+        match tp {
```

Comment:
It should be possible to do a simple comparison, as above.  Noting that in this case `self.initial_source_cid` probably should be `ConnectionId` rather than `Option<ConnectionId>`.

Reply:
done